### PR TITLE
feat: mobile main hud

### PR DESCRIPTION
**Pull request by kuruk-mm**
- Change Icons in the HUD
- Screen scaling. The base resolution for the UI is 1280x720, and if you change the size, the UI will resize to match that size.
- Menu navigation UI improved, header increased the size
- Joystick now works in an left-bottom area of the screen
- Chat only is shown when it's used (on the future, we are going to implement last 5 messages notifications above it)
- Implement SafeAreaHUD, an area that the "interactive" UI must be drawn. This is for avoiding physicals parts in the Mobile phone (like a camera, or some parts of it) that you cannot interact

### CODE REVIEW

File: .github/workflows/android-build.yml [hunk 1]
```diff
@@ -4,6 +4,10 @@ on:
   push:
     branches:
       - main
+  pull_request:
+  release:
```
Comment (leanmendoza) [inline .github/workflows/android-build.yml:8]: we're not using github release pipeline

File: godot/.godot/global_script_class_cache.cfg [hunk 2]
```diff
@@ -107,6 +107,12 @@ list=Array[Dictionary]([{
 "language": &"GDScript",
 "path": "res://src/logic/rust_http_requester_wrapper.gd"
 }, {
+"base": &"MarginContainer",
+"class": &"SafeMarginContainer",
+"icon": "",
+"language": &"GDScript",
+"path": "res://src/ui/components/safe_margin_container/SafeMarginContainer.gd"
```
Comment (leanmendoza) [inline godot/.godot/global_script_class_cache.cfg:114]:
```suggestion
"path": "res://src/ui/components/safe_margin_container/safe_margin_container.gd"
```

File: godot/src/logic/player/player.gd [hunk 3]
```diff
@@ -145,7 +145,7 @@ func _physics_process(delta: float) -> void:
 	if not is_on_floor():
 		velocity.y -= gravity * delta
 
-	elif Input.is_action_just_pressed("ia_jump"):
+	elif Input.is_action_pressed("ia_jump"):
```
Comment (leanmendoza) [inline godot/src/logic/player/player.gd:148]: 🙌

File: godot/src/main.gd [hunk 4]
```diff
@@ -14,19 +14,19 @@ func start():
 		print("Running from platform")
 		var resolution_manager = ResolutionManager.new()
 		resolution_manager.refresh_window_options()
-		resolution_manager.change_window_size(
-			get_window(), get_viewport(), Global.config.window_size
-		)
-		resolution_manager.change_resolution(get_window(), get_viewport(), Global.config.resolution)
-		resolution_manager.change_ui_scale(get_window(), Global.config.ui_scale)
+		#resolution_manager.change_window_size(
```
Comment (leanmendoza) [inline godot/src/main.gd:17]: dead code
this is still in the settings

File: godot/src/mobile/joystick/virtual_joystick.gd [hunk 5]
```diff
@@ -73,8 +73,12 @@ func _ready() -> void:
 		hide()
 
 
+func custom_input(event: InputEvent) -> void:
```
Comment (leanmendoza) [inline godot/src/mobile/joystick/virtual_joystick.gd:76]: can you detail how this works?

File: godot/src/ui/components/safe_margin_container/SafeMarginContainer.gd
Comment (leanmendoza) [inline godot/src/ui/components/safe_margin_container/SafeMarginContainer.gd:1]: change filename

**Review by leanmendoza — CHANGES_REQUESTED** [review]
Some minor reviews
Regarding the tests:
- Download the `snapshots.zip` from the CI summary and get the right snapshots to replace in `repo/tests/snapshots`
- Before replacing text-shape's snapshots, see what and why they're changing

File: .github/workflows/android-build.yml (hunk 1 above)
Comment (kuruk-mm) [reply .github/workflows/android-build.yml:8]: it's temp for copy paste

**Review by leanmendoza — APPROVED** [review]
Nice work, looks good to me 😄 
Just fix tests before merging

**Pull request merged**